### PR TITLE
Upgrade cocoa deps, add new instructions

### PR DESCRIPTION
**Pull request by kattrali**
* Updates bugsnag-cocoa version
* Add releasing instructions for updating vendored lib
* Resolves issue where callback severity reason is being collected as handled

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,7 +39,18 @@ to run the example projects.
...
+
+To upgrade bugsnag-cocoa, prepare an updated version of bugsnag-cocoa, then copy
+the Source and iOS directories. Assuming bugsnag-cocoa is located in the same
+directory as bugsnag-react-native:
+
+```
+cp -r ../bugsnag-cocoa/Source/* cocoa/vendor/bugsnag-cocoa/Source
+cp -r ../bugsnag-cocoa/iOS/* cocoa/vendor/bugsnag-cocoa/iOS
+```
+
```
Comment (fractalwrench) [inline CONTRIBUTING.md]: The iOS xcodeproj will also need to be copied over (especially if we're adding/removing source files)

File: cocoa/vendor/bugsnag-cocoa/Source/BugsnagHandledState.m [hunk 2]
```diff
@@ -116,8 +120,9 @@ + (NSString *)stringFromSeverityReason:(SeverityReasonType)severityReason {
             return kPromiseRejection;
         case UserSpecifiedSeverity:
             return kUserSpecifiedSeverity;
+        case LogMessage:
+            return kLogGenerated;
         case UnhandledException:
-        default:
             return kUnhandledException;
```
Comment (fractalwrench) [inline cocoa/vendor/bugsnag-cocoa/Source/BugsnagHandledState.m]: Should we use `NSParameterAssert` for the default case here?

Comment (kattrali) [reply cocoa/vendor/bugsnag-cocoa/Source/BugsnagHandledState.m]: If there's no default case, there's a build warning if any cases are missing. The change here was to specify every case so its not needed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kattrali) [reply CONTRIBUTING.md]: Isn't that in `iOS/`?

Comment (fractalwrench) [reply CONTRIBUTING.md]: Yes - my bad!

Comment (kattrali) [reply CONTRIBUTING.md]: No worries, I had to check!

**Pull request merged**